### ISSUE TITLE
[java] False positive of MissingStaticMethodInNonInstantiatableClass with Lombok Builder on Constructor

### ISSUE THREAD

**Issue by mmoehring** [pmd/pmd#1488]
**Affects PMD Version:** 6.90

**Rule:** MissingStaticMethodInNonInstantiatableClass

**Description:**
Using Lomboks Builder annotation on a private constructor triggers MissingStaticMethodInNonInstantiatableClass altough the static builder() method is generated and can be used normally.

**Code Sample demonstrating the issue:**

```
package org.example.foo;

import lombok.Builder;

/**
 * TestClass.
 */
public class TestClass {
    @Builder
    private TestClass(final String foo, final Integer bar) {
    }
}
```

**Running PMD through:** *[Gradle 3.5]*

**Comment by kdebski85** [comment]
I added a PR https://github.com/pmd/pmd/pull/5109 to fix the issue.

As a workaround, you can add `lombok.Builder` to `annotations`:

```xml
    <rule ref="category/java/errorprone.xml/MissingStaticMethodInNonInstantiatableClass">
        <properties>
            <property name="annotations" value="org.springframework.beans.factory.annotation.Autowired,javax.inject.Inject,com.google.inject.Inject,lombok.Builder" />
        </properties>
    </rule>
```
